packages/playground-examples/copy/en/4-7/New JS Features/Improved Function Inference.ts

```diff
@@ -0,0 +1,50 @@
+//// { "compiler": { "ts": "4.7.3" } }
+// Wth TypeScript 4.7, there are improvements in how types
+// are inferred from typed function calls.
+
+// Take this function 'cache', it takes one type parameter, 'Type'
+// and one function argument, which contains two functions.
+
+declare function cache<Type>(arg: { add: (n: string) => Type; process: (x: Type) => void }): void;
+
+// When TypeScript is trying to infer the type of 'Type', it
+// has two potential places to look:
+//
+// - the return of 'add'
+// - the first parameter of 'remove'
+
+// TypeScript's code inference now takes more these cases into
+// account with more code styles, and also handle cases where the
+// types depend on each other.
+
+cache({
+  add: n => n,
```

Comment:
Why is this called `add`?

Reply:
I guess it could be `insert` instead? It's adding something to a cache